wpilibj/src/main/java/edu/wpi/first/wpilibj/controller/LTVUnicycleController.java

```diff
@@ -117,36 +117,45 @@ public void setTolerance(final Pose2d poseTolerance) {
    * <p>The reference pose, linear velocity, and angular velocity should come
    * from a {@link Trajectory}.
    *
-   * @param currentPose                   The current position of the robot.
-   * @param poseRef                       The desired pose of the robot.
-   * @param linearVelocityRefMetersPerSec The desired linear velocity of the robot.
-   * @param angularVelocityRefRadPerSec   The desired angular velocity of the robot.
+   * <p>The current linear velocity of the chassis can be found from a
+   * DifferentialDriveWheelSpeeds object using
+   * {@link edu.wpi.first.wpilibj.kinematics.DifferentialDriveWheelSpeeds#toLinearChassisVelocity()}
+   * .
+   *
+   * @param currentPose                       The current position of the robot.
+   * @param currentLinearVelocityMetersPerSec The current linear velocity of the robot.
+   *                                          this can be determined by averaging the
+   *                                          measured left and right wheel velocities.
+   * @param poseRef                           The desired pose of the robot.
+   * @param linearVelocityRefMetersPerSec     The desired linear velocity of the robot.
+   * @param angularVelocityRefRadPerSec       The desired angular velocity of the robot.
    * @return The next calculated output.
    */
-  public ChassisSpeeds calculate(Pose2d currentPose, Pose2d poseRef,
+  @SuppressWarnings("LocalVariableName")
+  public ChassisSpeeds calculate(Pose2d currentPose,
+                                 double currentLinearVelocityMetersPerSec,
+                                 Pose2d poseRef,
                                  double linearVelocityRefMetersPerSec,
                                  double angularVelocityRefRadPerSec) {
-
     m_poseError = poseRef.relativeTo(currentPose);
 
-    var kx = m_K0.get(0, 0);
-    var ky0 = m_K0.get(1, 1);
-    var ky1 = m_K1.get(1, 1);
-    var ktheta1 = m_K1.get(1, 2);
+    if (currentLinearVelocityMetersPerSec == 0.0) {
+      currentLinearVelocityMetersPerSec = 1e-9;
+    } else if (Math.abs(currentLinearVelocityMetersPerSec) < 1e-9) {
+      currentLinearVelocityMetersPerSec = 1e-9 * Math.signum(currentLinearVelocityMetersPerSec);
+    }
```

Comment:
There should be a comment explaining this code's existence, like this: https://github.com/frc3512/Robot-2020/blob/master/src/main/cpp/controllers/DrivetrainController.cpp#L241-L242